examples/experiments/grounded_program_synthesis/lang.py

```diff
@@ -0,0 +1,319 @@
+import random
+import copy
+from pprint import pprint
+from tqdm import tqdm
+import json
+
+
+def init_random_input(len_range:int=5,value_gen=5)->list:
+    len_gen = random.randint(2, len_range+1)
+    value_range = list(range(-value_gen, value_gen+1))
+    output = []
+    for index in range(len_gen):
+        value_gen = random.choice(value_range)
+        output.append(value_gen)
+    return output
+
+
```

Comment:
Could you comment this file more? Maybe to separate the sections and an explanation at the top for the structure of the language/interpreter

Reply:
Strong agree. The documentation here is not clear.

Reply:
Adding more documentation, thanks for the review.

Reply:
Added more documentation on the sampler.